### PR TITLE
Add GemmProblemTypeV2, GemmEpilogueV2, and GemmInputsV2 for Backward Compatibility

### PR DESCRIPTION
**Pull request by KKyang**

### CODE REVIEW

File: library/include/hipblaslt-ext.hpp [hunk 1]
```diff
@@ -117,6 +120,56 @@ namespace hipblaslt_ext
...
+
+        HIPBLASLT_EXPORT                 GemmEpilogueV2(const GemmEpilogueV2& epilogue);
+        HIPBLASLT_EXPORT GemmEpilogueV2& operator=(const GemmEpilogueV2& epilogue);
+
+        HIPBLASLT_EXPORT                 GemmEpilogueV2(GemmEpilogueV2&& epilogue);
+        HIPBLASLT_EXPORT GemmEpilogueV2& operator=(GemmEpilogueV2&& epilogue);
+
+        HIPBLASLT_EXPORT void
+            setMode(hipblasLtEpilogue_t mode); //!< Set the mode of epilogue. Default is gemm.
+        HIPBLASLT_EXPORT void setBiasDataType(
```
Comment (jichangjichang) [inline library/include/hipblaslt-ext.hpp:188]: It is better to return status for SET APIs

File: library/include/hipblaslt-ext.hpp [hunk 2]
```diff
@@ -117,6 +120,56 @@ namespace hipblaslt_ext
...
+        HIPBLASLT_EXPORT void setAuxLeadingDimension(
+            int auxLeadingDimension); //!< Set the aux leading dimension. Only works if mode is set to aux related epilogues.
+        HIPBLASLT_EXPORT void setAuxBatchStride(
+            int auxBatchStride); //!< Set the aux batch stride. Only works if mode is set to aux related epilogues.
+        HIPBLASLT_EXPORT void setScalingAType(
+            int scalingAType); //!< 0 is scalar, 1 is vector. Only works if DataTypeA = DataTypeB = FP8.
+        HIPBLASLT_EXPORT void setScalingBType(
+            int scalingBType); //!< 0 is scalar, 1 is vector. Only works if DataTypeA = DataTypeB = FP8.
+
+        HIPBLASLT_EXPORT hipblasLtEpilogue_t
```
Comment (jichangjichang) [inline library/include/hipblaslt-ext.hpp:200]: also needs return status for Get API and return actual user requested data in the pointer

Comment (KKyang) [reply library/include/hipblaslt-ext.hpp:200]: Too C style, the private members are just structures, It returns default values if not set.

File: library/include/hipblaslt-ext.hpp (hunk 1 above)
Comment (KKyang) [reply library/include/hipblaslt-ext.hpp:188]: SUCCUESS is the only option, is this necessary?

File: library/include/hipblaslt-ext.hpp (hunk 2 above)
Comment (KKyang) [reply library/include/hipblaslt-ext.hpp:200]: Different from the C API, C API might fail cause the API is

```
getAttribute(ATTRIBUTE_ENUM, &output_data);
```

User might give an invalid enum, or a null output.

However, Cpp APIs are just get and set wrapping around the pimpl class.

File: library/src/amd_detail/hipblaslt-ext.cpp [hunk 3]
```diff
@@ -360,8 +669,54 @@ namespace hipblaslt_ext
...
+                                     int64_t          ldb,
+                                     int64_t          ldc,
+                                     int64_t          ldd,
+                                     int64_t          strideA,
+                                     int64_t          strideB,
+                                     int64_t          strideC,
+                                     int64_t          strideD,
+                                     GemmEpilogueV2&  epilogue,
+                                     GemmInputsV2&    inputs,
+                                     GemmProblemType& problemtype)
```
Comment (jichangjichang) [inline library/src/amd_detail/hipblaslt-ext.cpp]: Could we support GemmProblemTypeV2 as well?

Comment (KKyang) [reply library/src/amd_detail/hipblaslt-ext.cpp]: Will this change in the future? If highly possible, we should change this too.

Comment (jichangjichang) [reply library/src/amd_detail/hipblaslt-ext.cpp]: Original setProblem: epilogev1, inputv1, problemtypev
this PR add setProblem: epilogev2, inputv2, problemtypev1
setProblem in the future: epilogev2, inputv2, problemtypev2
Then we will have 3 versions of SetProblem need to take care in the future.
It is not must but better to change it together this time.

File: library/include/hipblaslt-ext.hpp (hunk 2 above)
Comment (jichangjichang) [reply library/include/hipblaslt-ext.hpp:200]: OK

**Comment by jichangjichang** [comment]
fix documentation
https://advanced-micro-devices-demo--919.com.readthedocs.build/projects/hipBLASLt/en/919/ext-reference.html#gemmepiloguev2

**Comment by KKyang** [comment]
> fix documentation https://advanced-micro-devices-demo--919.com.readthedocs.build/projects/hipBLASLt/en/919/ext-reference.html#gemmepiloguev2

Fixed

**Review by jichangjichang — APPROVED** [review]
TODO: GemmProblemTypeV2 Support ComputeInputTypeA/B in the future

**Comment by KKyang** [comment]
[tox.txt](https://github.com/user-attachments/files/16450697/tox.txt)
[test.txt](https://github.com/user-attachments/files/16450757/test.txt)

gfx90a tox/hipblaslt-test local test passed

**Review by hcman2 — APPROVED** [review]

**Pull request merged**